source/operators/captions/webcaptions.py

```diff
@@ -0,0 +1,1016 @@
+                self.operator_object.update_workflow_status("Error")
+                self.operator_object.add_workflow_metadata(
+                    WebCaptionsError="Unable to store srt captions file {e}".format(e=response))
+                raise MasExecutionError(self.operator_object.return_output_object())
+
+    def TranslateWebCaptions(self, inputCaptions, sourceLanguageCode, targetLanguageCodes, terminology_names=[], parallel_data_names=[]):
+
+        marker = self.marker
+
+        try:
+
+            translate_role = os.environ['translateRole']
+            asset_id = self.operator_object.asset_id
+            workflow_id = self.operator_object.workflow_execution_id
+            translate_job_name = "MIE_"+asset_id+"_"+workflow_id
+
+            # Convert WebCaptions to text with marker between caption lines
+            inputEntries = map(lambda c: c["caption"], inputCaptions)
+            inputDelimited = marker.join(inputEntries)
+
+            transcript_storage_path = dataplane.generate_media_storage_path(asset_id, workflow_id)
+            bucket = transcript_storage_path['S3Bucket']
+            translation_input_path = transcript_storage_path['S3Key']+"webcaptions_translate_input/"
+            translation_input_uri = 's3://'+bucket+"/"+translation_input_path
+            translation_output_path = transcript_storage_path['S3Key']+"webcaptions_translate_output/"
+            translation_output_uri = 's3://'+bucket+"/"+translation_output_path
+            key = translation_input_path+"transcript_with_caption_markers.txt"
+
+
+            print("put object {} {}".format(bucket, key))
+            s3.put_object(Bucket=bucket, Key=key, Body=inputDelimited)
+
+            print("create translate output folder if it doesn't exist")
+            dummy_key = translation_output_path+"/"+"foo"
+            s3.put_object(Bucket=bucket, Key=dummy_key, Body="foo")
+
+            print("Translate inputs")
+            print("translation_input_uri {}".format(translation_input_uri))
+            print("translation_output_uri {}".format(translation_output_uri))
+            print("translate_role {}".format(translate_role))
+
+            # Kick off a job for each input language
+            translate_jobs = []
+            for targetLanguageCode in targetLanguageCodes:
+                print("Starting translation to {}".format(targetLanguageCode))
+                # Even though the API takes a list of targets, Translate only supports
+                # a list of 1 or less
+
+                # Set the job name to avoid creating the same job multiple time if
+                # we retry due to an error.
+                singletonTargetList = []
+                singletonTargetList.append(targetLanguageCode)
+                job_name = "MIE_"+asset_id+"_"+workflow_id+"_"+targetLanguageCode
+                print("JobName: {}".format(job_name))
+
+                terminology_name = []
+                if len(terminology_names) > 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`.

 [Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)

Similar issue at line numbers 450, 456, and 465.